packages/berry-pnpify/sources/PnPRootLocator.ts

```diff
@@ -0,0 +1,143 @@
+import * as fs from 'fs';
```

Comment:
Be careful when importing `fs` - we usually work with normalized paths, and they aren't compatible with the classic `fs`. We use `import {xfs} from '@berry/fslib'` instead (which is an instance of `NodeFS`).

Reply:
@arcanis My rationale here was, that `node_modules` emulation layer will get paths from external tools, not from code which is within `berry`. Does this makes sense?

Reply:
Hm I guess 🤔

Fwiw, [the zipfs emulation layer](https://github.com/yarnpkg/berry/blob/master/packages/berry-pnp/sources/loader/applyPatch.ts#L262) (which also accepts native paths by necessity) still uses the fslib together with a [PosixFS](https://github.com/yarnpkg/berry/blob/master/packages/berry-fslib/sources/PosixFS.ts) instance which enables support for both portable paths and native paths (it transparently converts portable paths back and forth before forwarding them to the "real" fs).